src/main/java/duke/command/ByeCommand.java

```diff
@@ -0,0 +1,37 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents a command that when executed prepares the application for exiting.
+ */
+public class ByeCommand extends Command {
+    /**
+     * Prompts <code>ui</code> to show goodbye message to user.
+     * After that, gets the formatted task list from <code>taskList</code> 
+     * and directs <code>storage</code> to store it into disk.
+     *
+     * @param ui user interface of the application.
+     * @param taskList task list of the application.
+     * @param storage disk storage of the application.
+     * @throws DukeException when an exception is thrown in the process of executing this command.
+     */
+    @Override
+    public void execute(Ui ui, TaskList taskList, Storage storage) throws DukeException {
+        ui.sayBye();
+        storage.saveToFile(taskList.formatAsFileData());
+    }
+
+    /**
+     * Checks if this command is a goodbye command.
+     * 
+     * @return true.
+     */
+    @Override
+    public boolean isBye() {
+        return true;

```

Comment:
an alternative(that i saw from someone else's code) you can consider is:
```
return this instanceof ByeCommand;
```

in the abstract Command class.
That may help readers understand better, but it could be less efficient.
What do you think about this?

Reply:
I think the alternative is fine also, but I would still prefer overriding it in ByeCommand class because Command class knows nothing about its subclass, and I personally feel that writing code in superclass that refers to subclass is a bit weird. Just personal preferences I guess 😀

Reply:
that makes a very good point. I am convinced this is better :)